### DISCUSSION TITLE
v2.12.0

### DISCUSSION THREAD

**Discussion by sojan-official** [chatwoot/chatwoot#6104] [Releases & Announcements]
## Changelog

- Message Delivery and read status for Whatsapp Channel
- Ability to view unattended conversations
- Ability to create and configure agent bots from UI
- Ability to mark a conversation as unread
- Ability to lock conversations to a single thread for supported channels
- Ability to use postfix as the mailer in self-hosted installations
- Ability to search in emoji selector
- Allow wildcard URLs in campaigns
- Allow users to disable marking offline automatically
- Numerous bug fixes and enhancements
- New languages: Latvian

###
Thanks to @clairton, @giquieu, @jordan-brough, @civilcoder55, @smartdev58  for the contributions

<hr /><em>This discussion was created from the release <a href='https://github.com/chatwoot/chatwoot/releases/tag/v2.12.0'>v2.12.0</a>.</em>

**Comment by bernapastorini** [comment]
Hi! Where i can get more documentation about the language used to create Agent Bots from UI? Thanks

**Reply by pranavrajs** [reply]
https://www.csml.dev/

**Reply by pranavrajs** [reply]
It's CSML - the bot programming language.

**Comment by glauberssoutodossantos** [comment]
Hi @pranavrajs ,

does this feature works for 360Dialog too?

`Message Delivery and read status for Whatsapp Channel`

I test without sucess :(

**Reply by srvtotaldev** [reply]
It's not working with WhatsApp Cloud API either.

**Comment by jochenstu** [comment]
I am not seeing the new features in a self-hosted installation after running cwctl --upgrade (although the installation [seemed to go through just fine](https://gist.github.com/jochenstu/aaa289c74baa17191913f5ea91b4af6e)). Am I missing something here?

**Comment by anihawkins** [comment]
Same here! Not seeing the new features on my self hosted installation after using cwctl --upgrade

**Reply by grupale** [reply]
Hi! I've just upgrade to 2.12.1 with ctwctl tool and I cant see the agent bot feature in UI either

**Reply by pranavrajs** [reply]
@grupale Agentbot feature requires a couple of configuration. The docs are still not up as it is beta now.

**Comment by sojan-official** [comment]
## 2.12.0 Dev Changelog

### API Endpoints

#### Additions
- `/api/v1/accounts/{account.id}/conversations/:id/unread` [POST]
  -  https://github.com/chatwoot/chatwoot/pull/5924
  - Ability for agents to mark conversations as unread

- `/api/v1/profile/auto_offline` [POST]
  - https://github.com/chatwoot/chatwoot/pull/6079
  - Ability for users to disable the auto offline by system


#### Deletions
- none

---

### Other Changes

- none
